cdap-app-fabric/src/main/java/co/cask/cdap/templates/AdapterSpecification.java

```diff
@@ -134,11 +148,14 @@ public static Builder builder(String name, String template) {
     private Resources resources;
     private JsonElement config;
 
-    public Builder(String name, String template) {
+    public Builder(String name, String template, Id.Program program) {
```

Comment:
Is template name always the application name? If that's the case, if you have `Id.Program`, you don't need `template`, since it'd be the same as `program.getApplicationId()`. @albertshau Can you confirm?